### PR TITLE
Fixed Synchronization with leo_manager nodes

### PR DESCRIPTION
**Pull request by windkit**
### Description

Current Implementation of synchronizing bucket metadata with manager nodes has some problems.
1. If the master leo_manager is down, every operations would have to wait for RPC timeout and the last synchronized time is not updated.
2. Only the first one (master) leo_manager is contacted
### Related Issue

https://github.com/leo-project/leofs/issues/438

### CODE REVIEW

File: src/leo_s3_bucket.erl [hunk 1]
```diff
@@ -880,20 +880,47 @@ find_bucket_by_name_1(Bucket, DB, Providers) ->
...
+                  _ ->
+                      Value
+              end,
+    case Value_1 of
+        null ->
+            not_found;
+        not_found ->
+            not_found;
+        _ ->
+            NewBucketVal = Value_1#?BUCKET{last_synchroized_at = leo_date:now()},
```
Comment (mocchira) [inline src/leo_s3_bucket.erl:903]: Get to error if `Value_1` matches `{error, any()}`.
Need to handle separately.

File: src/leo_s3_bucket.erl [hunk 2]
```diff
@@ -880,20 +880,47 @@ find_bucket_by_name_1(Bucket, DB, Providers) ->
...
+                      not_found;
+                  _ ->
+                      Value
+              end,
+    case Value_1 of
+        null ->
+            not_found;
+        not_found ->
+            not_found;
+        _ ->
```
Comment (mocchira) [inline src/leo_s3_bucket.erl:902]: When `Value_1` matches `{error, any()}` due to timeout or badrpc,
It'd be good to log with warn level for informing admins that sync with leo_manager failed.

File: src/leo_s3_bucket.erl [hunk 3]
```diff
@@ -903,35 +930,33 @@ find_bucket_by_name_2(Bucket, DB, Node, Value) ->
...
+        {badrpc, Reason} ->
+            {error, Reason}
+    end,
+    case Ret of
+        {error, Cause} ->
+            error_logger:warning_msg("~p,~p,~p, Synchronization of Bucket Data Failed, Node:~p, Reason:~p~n",
+                                     [{module, ?MODULE_STRING}, 
+                                      {function, "find_bucket_by_name_2/3"},
+                                      {line, ?LINE},
+                                      Node, Cause]);
```
Comment (mocchira) [inline src/leo_s3_bucket.erl:954]: @windkit 
Must bind `Ret` with `{error, Cause}`.
if not, retrying against leo_manager slave will not occur.

Comment (windkit) [inline src/leo_s3_bucket.erl:954]: It is bind at line 932.

**Comment by mocchira** [comment]
@windkit right.
Thank you for your cooperation.

**Pull request merged**